src/librbd/internal.cc

```diff
@@ -1989,6 +1989,39 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     return 0;
   }
 
+  int snap_set(ImageCtx *ictx, const cls::rbd::SnapshotNamespace &snap_namespace,
```

Comment:
Not needed -- what I was implying was to add a new overload to ```ImageState::snap_set``` that just takes the snap_id, but it doesn't need to be part of this PR. In general, I am trying to (slowly) remove code from ```internal.cc```.